### PR TITLE
Remove StoreManagerInterface and use helper from Nosto_Tagging module instead

### LINKED ISSUE #101: Refactor usage of storeManager in the plugin
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
Use a helper method to get the store instead of calling `storeManager->getStore` in different places.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by dairbuirabass**
## Description
`StoreManagerInterface` usage is removed from the module and replaced with a `Nosto\Tagging\Helper\Scope` helper. Updated all methods and return types. Removed instance checks and phan escapes.

## Related Issue
Closes #101 

## Motivation and Context
Reduce direct usage of StoreManagerInterface.

## How Has This Been Tested?
Locally

## Checklist:
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] I have assigned the correct milestone or created one if non-existent.
- [x] I have correctly labeled this pull request.
- [x] I have linked the corresponding issue in this description.
- [x] I have updated the corresponding Jira ticket.
- [x] I have requested a review from at least 2 reviewers
- [x] I have checked the base branch of this pull request
- [x] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: Helper/Data.php [hunk 1]
```diff
@@ -93,53 +92,53 @@ public function __construct(
     /**
      * Returns if category sorting is enabled
      *
-     * @param StoreInterface|null $store the store model or null.
+     * @param Store|null $store the store model or null.
      * @return bool the configuration value
      */
-    public function isCategorySortingEnabled(StoreInterface $store = null)
+    public function isCategorySortingEnabled(Store $store = null)
```
Comment (mridang) [inline Helper/Data.php:98]: Why change from the interface to the concrete type?

Comment (dairbuirabass) [reply Helper/Data.php:98]: Scope helper returns concrete type

File: Block/CategoryMapping.php [hunk 2]
```diff
@@ -63,50 +59,44 @@ class CategoryMapping extends Template
...
-        try {
-            $store = $this->storeManager->getStore();
-            if ($store instanceof Store) {
-                $array = $this->getMagentoCategories($store);
-            }
-        } catch (NoSuchEntityException $e) {
-            $this->logger->exception($e);
-        }
-
+        $store = $this->nostoHelperScope->getStore();
```
Comment (mridang) [inline Block/CategoryMapping.php:98]: I don't think the `storeId` is available here so let's add a comement.

Comment (dairbuirabass) [reply Block/CategoryMapping.php:98]: It fetches current store if no storeId is given. What comment should I leave?

Comment (olsi-qose) [reply Block/CategoryMapping.php:98]: Yes, but if there's the possibility to pass the store id then we should pass it. If there's no way to get the store id, just put a comment.

**Review by olsi-qose — CHANGES_REQUESTED** [review]

File: Helper/Data.php (hunk 1 above)
Comment (dairbuirabass) [reply Helper/Data.php:98]: Changing the return type in the Helper will also require changes to the main module. Should I open parallel PR to this one?

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@ All notable changes to this project will be documented in this file. This projec
 ### 3.3.0-rc7
 * Fix search page bug for MySQL and ElasticSearch
 * Introduce possibility to define page size through DI
+* Replaced direct usage of StoreManagerInterface with a helper
```
Comment (olsi-qose) [inline CHANGELOG.md:6]: Remove this line, it's not included in RC-7

**Review by olsi-qose — CHANGES_REQUESTED** [review]

**Review by olsi-qose — APPROVED** [review]

**Pull request merged**